### ISSUE TITLE
Random Mix UI - "Play" button not working?

### ISSUE THREAD

**Issue by beatniq** [CDrummond/lms-material#969]
Hi,

Firstly, love your work on this LMS skin. You've done an incredible job with the UX for years and still keep finding new ways to make it better - awesome!

This is minor thing, but thought I'd raise it in case anyone else is experiencing the same. With the new Random Mix UI improvements, I have a strange issue whereby if I click the "play" button on an existing Random Mix, or similarly open the menu dialogue by clicking a Random Mix title and then choose, "Play now" - nothing happens, and I get a new line in my server.log as follows:

`Slim::Plugin::RandomPlay::Mixer::findAndAdd (50) Warning: Didn't get a valid object for findAndAdd()!`

But if I instead choose, "Edit" and then click the "Start" button in the resulting modal window, everything works as I would expect.

Does the "Start" action call a different function behind the scenes to the "Play" buttons? Could this be something I've misconfigured anywhere else?

Any pointers/suggestions would be welcome. Thanks.

![Screenshot 2024-10-25 at 11 21 46](https://github.com/user-attachments/assets/cd8d0ffb-4ec5-4c09-80e8-f9581eb6d8fc)

**Comment by CDrummond** [comment]
Is this all genres?

**Comment by beatniq** [comment]
No, I'd handpicked 3 on this occasion. But just tried it again with a single Genre and hit the same issue.

**Comment by CDrummond** [comment]
Odd. The code for the play button is different - it sends a single command to the LMS server, which read the stored mix  file and then sends the commands. For 5.4.2 I've changed the code so that the play action will now load the mix dialog and immediately start the mix - without actually showing the dialog. Therefore, the same code path should be used...

**Comment by beatniq** [comment]
Great, thanks. I'll be sure to report back after I've had the chance to try 5.4.2 👍

**Comment by beatniq** [comment]
Confirmed fixed in `5.4.2` 👍 

Thanks.
